include/subgroup/tile/impl/fma_xe.hpp

```diff
@@ -24,6 +24,129 @@
 namespace gpu::xetla::subgroup {
 
 /// @brief Is the tile mma operation functor, specialized for Xe and fpu engine.
+template <
+    typename matDst_t_,
+    typename matSrc_t_,
+    typename matAcc_t_,
+    typename matB_t_,
+    typename matA_t_,
+    gpu_arch arch_tag_>
+struct tile_fma_t {
+  using matA_t = matA_t_;
+  using matB_t = matB_t_;
+  using matSrc_t = matSrc_t_;
+  using matDst_t = matDst_t_;
+  using matAcc_t = matAcc_t_;
+  using dtype_a = typename matA_t::dtype;
+  using dtype_b = typename matB_t::dtype;
+  using dtype_src = typename matSrc_t::dtype;
+  using dtype_dst = typename matDst_t::dtype;
+
+  using register_attr =
+      typename arch_attr_t<arch_tag_>::template register_attr<>;
+
+  static constexpr uint32_t a_tile_size_y = matA_t::tile_size_y;
+  static constexpr uint32_t a_tile_size_x = matA_t::tile_size_x;
+  static constexpr uint32_t a_tile_elems = matA_t::tile_elems;
+  static constexpr uint32_t a_block_size_y = matA_t::block_size_y;
+  static constexpr uint32_t a_block_size_x = matA_t::block_size_x;
+  static constexpr uint32_t a_block_elems = matA_t::block_elems;
+
+  static constexpr uint32_t b_tile_size_x = matB_t::tile_size_x;
+  static constexpr uint32_t b_tile_size_y = matB_t::tile_size_y;
+  static constexpr uint32_t b_tile_elems = matB_t::tile_elems;
+  static constexpr uint32_t b_block_size_x = matB_t::block_size_x;
+  static constexpr uint32_t b_block_size_y = matB_t::block_size_y;
+  static constexpr uint32_t b_block_elems = matB_t::block_elems;
+
+  static constexpr uint32_t tile_size_m = a_tile_size_y;
+  static constexpr uint32_t tile_size_k = a_tile_size_x;
+  static constexpr uint32_t tile_size_n = b_tile_size_x;
+  static constexpr uint32_t block_size_m = a_block_size_y;
+  static constexpr uint32_t block_size_k = a_block_size_x;
+  static constexpr uint32_t block_size_n = b_block_size_x;
+
+  static_assert(
+      a_tile_size_x == b_tile_size_y,
+      "matA tile k should match with matB tile k");
+  static_assert(
+      a_block_size_x == b_block_size_y,
+      "matA block k should match with matB block k");
+  static_assert(
+      b_block_size_y == matAcc_t::block_size_x,
+      "matA block k should match with matAcc block k");
+  static_assert(
+      b_block_size_x == matAcc_t::block_size_y,
+      "matb block n should match with matAcc block n");
+
+  static_assert(tile_size_m == 1, "matA tile m must be 1");
+  static_assert(a_block_size_y == 1, "matA block m must be 1");
+  __XETLA_API static void fma(
```

Comment:
why different naming?

```suggestion
  __XETLA_API static void mma(
```

Reply:
looks like this func only supports vec * matrix, is MMA appropriate

Reply:
looks like fma only support vec * matrix

Reply:
Just like zhe said
MMA is the product of matrices, while FMA only supports row-by-row multiplication of vec and matrix, and the output shape is the same as the matrix. An additional recude is required to get the final result. It can be understood as splitting the original MMA into two steps: executing FMA at k-dim and executing reduce at the end.